src/trio/testing/_raises_group.py

```diff
@@ -154,6 +154,66 @@ def _stringify_exception(exc: BaseException) -> str:
 _REGEX_NO_FLAGS = re.compile(r"").flags
 
 
+def _match_pattern(match: Pattern[str]) -> str | Pattern[str]:
+    """helper function to remove redundant `re.compile` calls when printing regex"""
+    return match.pattern if match.flags == _REGEX_NO_FLAGS else match
+
+
+def _check_match(match_expr: Pattern[str] | None, e: BaseException) -> str | None:
+    if match_expr is not None and not re.search(
+        match_expr,
+        stringified_exception := _stringify_exception(e),
+    ):
+        fail_reason = f"Regex pattern {_match_pattern(match_expr)!r} did not match {stringified_exception!r}"
+        if _match_pattern(match_expr) == stringified_exception:
+            fail_reason += "\nDid you mean to `re.escape()` the regex?"
```

Comment:
```suggestion
        if _match_pattern(match_expr) == stringified_exception != re.escape(match_expr):
            fail_reason += "\nDid you mean to `re.escape()` the regex?"
```
(maybe Pytest also needs to do this?)

Reply:
I .. don't think there is any pattern that will fail to `re.search` itself after being escaped. With `re.match` Claude managed to come up with an example, but as far as I can tell there is no way to make `re.search(re.escape(p), re.escape(p))` return True. Flags also can't help out, since that will not pass the equality check.

I could add it in case I'm mistaken though, and do the same when adding it upstream.